File: cx-organization-api/cxOrganization.Domain/RequestDtos/FileRequest/GetFileByUserIdRequest.cs
using cxOrganization.Domain.DomainEnums;
using System;

namespace cxOrganization.Domain.RequestDtos.FileRequest
{
    public class GetFileByUserIdRequest
    {
        public string SearchText { get; set; }

        public Guid UserGuid { get; set; }

        public int PageIndex { get; set; }

        public int PageSize { get; set; }

        public FileTarget FileTarget { get; set; }

        public OrderBy OrderBy { get; set; }

        public OrderByType OrderType { get; set; }
    }
}

File: cx-organization-spa/src/app/user-accounts/dtos/get-file-by-user-id-request.ts
import { FileTarget } from '../models/file-info.model';

export interface IGetFileByUserIdRequest {
  searchText?: string;
  userGuid?: string;
  pageIndex?: number;
  pageSize?: number;
  fileTarget: FileTarget;
  orderBy?: string;
  orderType?: string;
}

export class GetFileByUserIdRequest implements IGetFileByUserIdRequest {
  searchText?: string;
  userGuid?: string;
  pageIndex?: number;
  pageSize?: number;
  fileTarget: FileTarget;
  orderBy?: string;
  orderType?: string;

  constructor(data?: Partial<IGetFileByUserIdRequest>) {
    if (!data) {
      return;
    }

    this.searchText = data.searchText;
    this.userGuid = data.userGuid;
    this.pageIndex = data.pageIndex;
    this.pageSize = data.pageSize;
    this.fileTarget = data.fileTarget;
    this.orderBy = data.orderBy;
    this.orderType = data.orderType;
  }
}
